runtime/jcl/common/jclvm.c

```diff
@@ -300,7 +448,68 @@ hasConstructor(J9VMThread *vmThread, J9StackWalkState *state)
 	}
 }
 
+static J9HashTable *
+collectHeapStatistics(J9VMThread *vmThread) {
+	J9JavaVM *vm = vmThread->javaVM;
+	J9HashTable *hashTable = hashTableNew(
+			OMRPORT_FROM_J9PORT(vm->portLibrary),
+			J9_GET_CALLSITE(),
+			64,
+			sizeof(J9HeapStatisticsTableEntry),
+			sizeof(U_8*),
+			0,
+			J9MEM_CATEGORY_CLASSES,
+			heapStatisticsHashFn,
+			heapStatisticsHashEqualFn,
+			NULL,
+			vm
+	);
+
+	vm->memoryManagerFunctions->j9mm_iterate_all_objects(vmThread->javaVM,
```

Comment:
The return value should be checked for success.

Reply:
See https://docs.oracle.com/javase/8/docs/platform/jvmti/jvmti.html#jvmtiIterationControl
You can check which constant is returned. JVMTI_ITERATION_CONTINUE should mean success.

Reply:
The reason to check is because an OOM may have occurred and the hashtable may not be fully populated.